### PR TITLE
Add first draft of balenaOS masterclass

### PR DESCRIPTION
**Pull request by ZubairLK**
Signed-off-by: Zubair Lutfullah Kakakhel <zubair@balena.io>

### CODE REVIEW

File: README.md
Comment (xginn8) [inline README.md:40]:
```suggestion
- `resin-boot`: Contains the boot files
```

File: README.md
Comment (xginn8) [inline README.md:130]: Is there meant to be content here?

File: README.md
Comment (xginn8) [inline README.md:156]:
```suggestion
- `plymouth*.service`: daemon that manages the balenaOS logo on the screen when booted (splash screen)
```

File: README.md
Comment (xginn8) [inline README.md:301]:
```suggestion
## pretty-printing config.json.
```

File: README.md
Comment (xginn8) [inline README.md:357]:
```suggestion
This remounts the filesystem in read-write mode. Then you can just edit files.
```

File: README.md
Comment (xginn8) [inline README.md:425]:
```suggestion
Here is how you would search it if `CONFIG_SPI` is enabled in a device.
```

File: README.md
Comment (xginn8) [inline README.md:437]:
```suggestion
We can use docker to spin up a balenaOS container and run bash to poke around. Useful for various use-cases.
```

File: README.md
Comment (xginn8) [inline README.md:496]: To me this section is something we should be heavily discouraging (instead redirecting to https://github.com/balena-io-playground/configizer or similar), but maybe it's useful for debugging?

File: README.md
Comment (xginn8) [inline README.md:162]:
```suggestion
Some common uses:
```

File: README.md
Comment (xginn8) [inline README.md:37]:
```suggestion
# 1. Filesystem and partition layout
```

just to keep with other classes stylistically, ditto all the other headers beneath.

**Review by xginn8 — APPROVED** [review]
lgtm, it would be interesting to see how all the units fit together/depend on one another too (something like `systemd-analyze plot`)

**Comment by xginn8** [comment]
One other small note: you should add yourself to the CODEOWNERS @ZubairLK

**Comment by hedss** [comment]
Please don't merge until my review is up. I'm still awaiting Github to come back to me with an answer. If they don't today, I'll try doing smaller comment blocks :(

**Review by imrehg — DISMISSED** [review]
To block merge until Heds' comments to submit

**Comment by xginn8** [comment]
@imrehg since "requested changes" need to be specifically undone by their creator, I am happy to flip mine (and you can remove yours), such that we don't get stuck making progress here. Thoughts?

**Comment by hedss** [comment]
I've given up on Github coming back with an answer. I'm going to post screenshots of my review here:

screencapture-github-balena-io-projects-balenaos-masterclass-pull-1-files-2019-11-07-10_05_56
screencapture-github-balena-io-projects-balenaos-masterclass-pull-1-files-2019-11-07-10_05_56-2

**Comment by ZubairLK** [comment]
resolved by #2

**Pull request closed without merging**
